No-script datasets with ZIP files do not load

While investigating an issue on a Hub dataset, I have discovered the no-script datasets containing ZIP files do not load.

For example, that no-script dataset containing ZIP files, raises NonMatchingSplitsSizesError:
```python
In [2]: ds = load_dataset("sidovic/LearningQ-qg")

NonMatchingSplitsSizesError: [
  {
    'expected': SplitInfo(name='train', num_bytes=0, num_examples=188660, shard_lengths=None, dataset_name=None), 
    'recorded': SplitInfo(name='train', num_bytes=0, num_examples=0, shard_lengths=None, dataset_name='learning_q-qg')
  }, {
    'expected': SplitInfo(name='validation', num_bytes=0, num_examples=20630, shard_lengths=None, dataset_name=None), 
    'recorded': SplitInfo(name='validation', num_bytes=0, num_examples=0, shard_lengths=None, dataset_name='learning_q-qg')
  }, {
    'expected': SplitInfo(name='test', num_bytes=0, num_examples=18227, shard_lengths=None, dataset_name=None), 
    'recorded': SplitInfo(name='test', num_bytes=0, num_examples=0, shard_lengths=None, dataset_name='learning_q-qg')
  }
]
```

As another example, a no-script dataset containing just a (CSV)-ZIP file, raises a DatasetGenerationError:
```
>               num_examples, num_bytes = writer.finalize()

src/datasets/builder.py:1949: 

>           raise SchemaInferenceError("Please pass `features` or at least one example when writing data")
E           datasets.arrow_writer.SchemaInferenceError: Please pass `features` or at least one example when writing data

src/datasets/arrow_writer.py:598: SchemaInferenceError

The above exception was the direct cause of the following exception:

src/datasets/load.py:2143: in load_dataset
    builder_instance.download_and_prepare(
src/datasets/builder.py:954: in download_and_prepare
    self._download_and_prepare(
src/datasets/builder.py:1049: in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
src/datasets/builder.py:1813: in _prepare_split
    for job_id, done, content in self._prepare_split_single(

>           raise DatasetGenerationError("An error occurred while generating the dataset") from e
E           datasets.builder.DatasetGenerationError: An error occurred while generating the dataset

src/datasets/builder.py:1958: DatasetGenerationError
```

After investigating, I think this bug was introduced in this PR:
- #5972

Related to:
- https://huggingface.co/datasets/sidovic/LearningQ-qg/discussions/1

CC: @lhoestq

--- a/src/datasets/packaged_modules/__init__.py
+++ b/src/datasets/packaged_modules/__init__.py
@@ -60,5 +60,5 @@ _MODULE_TO_EXTENSIONS: Dict[str, List[str]] = {}
 for _ext, (_module, _) in _EXTENSION_TO_MODULE.items():
     _MODULE_TO_EXTENSIONS.setdefault(_module, []).append(_ext)
 
-_MODULE_TO_EXTENSIONS["imagefolder"].append(".zip")
-_MODULE_TO_EXTENSIONS["audiofolder"].append(".zip")
+for _module in _MODULE_TO_EXTENSIONS:
+    _MODULE_TO_EXTENSIONS[_module].append(".zip")

--- a/tests/fixtures/files.py
+++ b/tests/fixtures/files.py
@@ -289,7 +289,7 @@ def bz2_csv_path(csv_path, tmp_path_factory):
 
 @pytest.fixture(scope="session")
 def zip_csv_path(csv_path, csv2_path, tmp_path_factory):
-    path = tmp_path_factory.mktemp("data") / "dataset.csv.zip"
+    path = tmp_path_factory.mktemp("zip_csv_path") / "csv-dataset.zip"
     with zipfile.ZipFile(path, "w") as f:
         f.write(csv_path, arcname=os.path.basename(csv_path))
         f.write(csv2_path, arcname=os.path.basename(csv2_path))

--- a/tests/test_load.py
+++ b/tests/test_load.py
@@ -1458,3 +1458,12 @@ def test_load_dataset_with_storage_options_with_decoding(mockfs, image_file):
     ds = load_dataset("imagefolder", data_files=data_files, storage_options=mockfs.storage_options)
     assert len(ds["train"]) == 1
     assert isinstance(ds["train"][0]["image"], PIL.Image.Image)
+
+
+def test_load_dataset_without_script_with_zip(zip_csv_path):
+    path = str(zip_csv_path.parent)
+    ds = load_dataset(path)
+    assert list(ds.keys()) == ["train"]
+    assert ds["train"].column_names == ['col_1', 'col_2', 'col_3']
+    assert ds["train"].num_rows == 8
+    assert ds["train"][0] == {'col_1': 0, 'col_2': 0, 'col_3': 0.0}

--- a/tests/test_load.py
+++ b/tests/test_load.py
@@ -1464,6 +1464,6 @@ def test_load_dataset_without_script_with_zip(zip_csv_path):
     path = str(zip_csv_path.parent)
     ds = load_dataset(path)
     assert list(ds.keys()) == ["train"]
-    assert ds["train"].column_names == ['col_1', 'col_2', 'col_3']
+    assert ds["train"].column_names == ["col_1", "col_2", "col_3"]
     assert ds["train"].num_rows == 8
-    assert ds["train"][0] == {'col_1': 0, 'col_2': 0, 'col_3': 0.0}
+    assert ds["train"][0] == {"col_1": 0, "col_2": 0, "col_3": 0.0}

